services/QuillLMS/app/controllers/teachers/units_controller.rb

```diff
@@ -139,17 +139,31 @@ def destroy
   # :activity_id (in url)
   # :classroom_unit_id
   def score_info
-    completed = ActivitySession.where(
-      classroom_unit_id: params[:classroom_unit_id],
-      activity_id: params[:activity_id],
-      is_final_score: true,
-      visible: true
-    )
+    classroom_unit = ClassroomUnit.find(params[:classroom_unit_id])
```

Comment:
Since the exception clause below is being used for control flow, what about using a find_by and guard clause instead:

```
classroom_unit = ClassroomUnit.find_by(params[:classroom_unit_id])
return render json: { cumulative_score: 0, completed_count: 0 } if classroom_unit.nil?
```

Reply:
Not opposed.  Refactored.